flow/project.py

```diff
@@ -783,49 +952,54 @@ def np_callable(job):
         # Evaluate strings and callables for job:
         def evaluate(value):
             if value and callable(value):
-                return value(job)
+                return value(*jobs)
             elif isinstance(value, str):
-                return value.format(job=job)
+                return value.format(jobs=' '.join(map(str, jobs)))
             else:
                 return value
         return {key: evaluate(value) for key, value in directives.items()}
 
-    def _submit_cmd(self, entrypoint, ignore_conditions, job=None):
-        entrypoint = self._determine_entrypoint(entrypoint, dict(), job)
+    def _submit_cmd(self, entrypoint, ignore_conditions, jobs=None):
+        entrypoint = self._determine_entrypoint(entrypoint, dict(), jobs)
         cmd = "{} run -o {}".format(entrypoint, self.name)
-        cmd = cmd if job is None else cmd + ' -j {}'.format(job)
+        cmd = cmd if jobs is None else cmd + ' -j {}'.format(' '.join(map(str, jobs)))
         cmd = cmd if self.options is None else cmd + ' ' + self.options
         if ignore_conditions != IgnoreConditions.NONE:
             return cmd.strip() + ' --ignore-conditions=' + str(ignore_conditions)
         else:
             return cmd.strip()
 
-    def _run_cmd(self, entrypoint, operation_name, operation, directives, job):
+    def _run_cmd(self, entrypoint, operation_name, operation, directives, jobs):
         if isinstance(operation, FlowCmdOperation):
-            return operation(job).lstrip()
+            return operation(jobs).lstrip()
         else:
-            entrypoint = self._determine_entrypoint(entrypoint, directives, job)
-            return '{} exec {} {}'.format(entrypoint, operation_name, job).lstrip()
+            entrypoint = self._determine_entrypoint(entrypoint, directives, jobs)
+            return '{} exec {} {}'.format(
+                    entrypoint, operation_name, ' '.join(map(str, jobs))
+                ).lstrip()
 
     def __iter__(self):
         yield from self.operations.values()
 
     def __repr__(self):
         return "{type}(name='{name}', operations='{operations}', " \
-               "operation_directives={directives}, options='{options}')".format(
+               "operation_directives={directives}, options='{options}', " \
+               "flow_aggregate={flow_aggregate}, flow_select={flow_select})".format(
                    type=type(self).__name__,
                    name=self.name,
                    operations=' '.join(list(self.operations)),
                    directives=self.operation_directives,
-                   options=self.options)
+                   options=self.options,
+                   flow_aggregate=self.flow_aggregate,
+                   flow_select=self.flow_select)
 
-    def eligible(self, job, ignore_conditions=IgnoreConditions.NONE):
+    def eligible(self, jobs, ignore_conditions=IgnoreConditions.NONE):
```

Comment:
This is a great example of a user facing function that takes a list of jobs rather than any number of jobs. This can/will be very confusing for users. We need to make sure that we find a consistent API for users with this. This may mean hiding many more functions/methods.